### ISSUE TITLE
Clarification on Negative T1/2 and Clearance Values

### ISSUE THREAD

**Issue by anhtuannbk** [swansonk14/admet_ai#15]
Dear Kyle,

I have been using the ADMET-AI software to predict pharmacokinetic properties of compounds. However, I noticed that some predictions, specifically the half-life (T1/2) and clearance (CL), resulted in negative values. As these parameters are biologically expected to be positive, could you please clarify:

What could cause the model to produce negative values for T1/2 and clearance?
Is this a known issue related to out-of-domain predictions or data input errors?
Are there any steps I can take to avoid such results or verify their validity?
Thank you for your time and assistance. I look forward to your response.

Best regards,
Tuan

**Comment by swansonk14** [comment]
Hi Tuan,

Thank you for raising this issue!

There are unfortunately some cases where ADMET-AI will predict biologically infeasible values, such as a negative half-life. This is because for regression-valued datasets like half-life, the last layer of the underlying Chemprop graph neural network is a linear layer with an unrestricted range. Theoretically, the model should learn that the values are always positive, but given the limited sizes of most of our training sets, the model may still make mistakes and predict negative values, particularly on out-of-distribution molecules.

The best way to avoid this for now is to clip the predicted values post-hoc. This is actually already done on the ADMET-AI web interface. For example, you can clip any negative half-life value to 0. A negative value means the model believes the half-life of the molecule is smaller than for most other molecules, so setting the half-life to 0, or more realistically to a small positive number, is a reasonable approximation for what the model is trying to do. That said, perhaps an even safer thing to do would be to not trust negative predictions since it implies that the model is trying to generalize beyond the space of molecules it was trained on.

Best,
Kyle

**Comment by davide-grheco** [comment]
To comment on this: I'd rather have the negative values and decide what to do with them myself.
But I guess an option may be useful for users to enable such behavior if they find it better to only have sensed results.

**Comment by swansonk14** [comment]
@davide-grheco I agree! Right now, the clipping to 0 is done only for visualization purposes on the website. The raw negative values are preserved when downloading the results from the website or when running ADMET-AI locally.
